internal/buildinfo/version.go

```diff
@@ -8,17 +8,126 @@ import (
 	"time"
 
 	"github.com/blang/semver"
+	"github.com/loadsmart/calver-go/calver"
 	"github.com/superfly/flyctl/terminal"
 )
 
+type Version interface {
+	String() string
+	EQ(other Version) bool
+	Outdated(string) (bool, error)
+}
+
+type SemverVersion struct {
+	semver.Version
+}
+
+func ParseSemver(version string) (*SemverVersion, error) {
+	v, err := semver.ParseTolerant(version)
+	if err != nil {
+		return nil, err
+	}
+	return &SemverVersion{v}, nil
+}
+
+func (v SemverVersion) EQ(other Version) bool {
+	o, ok := other.(*SemverVersion)
+	if ok {
+		return v.Version.EQ(o.Version)
+	} else {
+		return false
+	}
+}
+
+func (v *SemverVersion) Outdated(other string) (bool, error) {
+	otherParsed, err := ParseSemver(other)
+	if err != nil {
+		return false, err
+	}
+	return v.LT(otherParsed.Version), nil
```

Comment:
Perhaps in SemverVersion's `Outdated` we should try to parse `other` as calver first, and on success unconditionally return true?